examples/offline_profile.py

```diff
@@ -89,50 +108,59 @@ def run_profile(context: ProfileContext, csv_output: Optional[str],
     with nm_profile() as prefill_prof:
         llm.llm_engine.step()  # First step is prefill
 
-    with nm_profile() as decode_prof:
-        llm.llm_engine.step()
+    decode_results_list = []
+    for _ in range(context.output_len - 1):
+        with nm_profile() as decode_prof:
+            llm.llm_engine.step()
+        decode_results_list.append(decode_prof.results)
 
     prefill_results = prefill_prof.results
-    decode_results = decode_prof.results
+    has_decode = len(decode_results_list) > 0
 
     print("=" * 80)
     print(f"= Prefill Model Table "
           f"(prompt_len={prompt_len}, batch_size={batch_size})")
     print("=" * 80)
     print()
     prefill_results.print_model_table()
-    print()
-    print("=" * 80)
-    print(f"= Decode Model Table "
-          f"(prompt_len={prompt_len}, batch_size={batch_size})")
-    print("=" * 80)
-    print()
-    decode_results.print_model_table()
+
+    if has_decode:
+        print()
+        print("=" * 80)
+        print(f"= Decode Model Table "
```

Comment:
perhaps's mention that this is just the summary of a single decode step.

Reply:
done!